Mathlib/Combinatorics/SimpleGraph/Connectivity/Subgraph.lean

```diff
@@ -238,6 +254,71 @@ theorem toSubgraph_adj_iff {u v u' v'} (w : G.Walk u v) :
     rw [← Subgraph.mem_edgeSet, ← hi.1, Subgraph.mem_edgeSet]
     exact toSubgraph_adj_getVert _ hi.2
 
+namespace IsPath
+
+lemma neighborSet_toSubgraph_startpoint {u v} {p : G.Walk u v}
+    (hp : p.IsPath) (hnp : ¬ p.Nil) : p.toSubgraph.neighborSet u = {p.snd} := by
+  have hadj1 := p.toSubgraph_adj_snd hnp
+  ext v
+  simp_all only [Subgraph.mem_neighborSet, Set.mem_insert_iff, Set.mem_singleton_iff,
+    SimpleGraph.Walk.toSubgraph_adj_iff, Sym2.eq, Sym2.rel_iff', Prod.mk.injEq, Prod.swap_prod_mk]
+  refine ⟨?_, by aesop⟩
+  rintro ⟨i, hl | hr⟩
+  · have : i = 0 := by
+      apply hp.getVert_injOn (by rw [Set.mem_setOf]; omega) (by rw [Set.mem_setOf]; omega)
+      aesop
+    aesop
+  · have : i + 1 = 0 := by
+      apply hp.getVert_injOn (by rw [Set.mem_setOf]; omega) (by rw [Set.mem_setOf]; omega)
+      aesop
+    contradiction
+
+lemma neighborSet_toSubgraph_endpoint {u v} {p : G.Walk u v}
+    (hp : p.IsPath) (hnp : ¬ p.Nil) : p.toSubgraph.neighborSet v = {p.penultimate} := by
+  simpa using IsPath.neighborSet_toSubgraph_startpoint hp.reverse
+      (by rw [Walk.not_nil_iff_lt_length, Walk.length_reverse]; exact
+        Walk.not_nil_iff_lt_length.mp hnp)
+
+lemma neighborSet_toSubgraph_internal {u} {i : ℕ} {p : G.Walk u v} (hp : p.IsPath)
+    (h : i ≠ 0) (h' : i < p.length) :
```

Comment:
No action required but often a lemma which consumes a natural number together with a hypothesis that it is non-zero, is better stated in terms of its successor.

Reply:
Really? `n` + `n \ne 0` is more syntactically general than `n + 1`. We've done the opposite change to what you're suggesting many times

Reply:
Yes, I'm aware. It just depends on the situation.

Reply:
Kept it as-is for now, but I'll keep it in mind for the future/refactoring if the other option turns out to be more useful.